### ISSUE TITLE
[Configuration Confusion] Confusion due to Repository Branch setup

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#6082]
**Describe the bug**
I saw that there was a bunch of merges, so I quickly went to my git and updated my master copy to update things, and was confused when git only listed two changes, not seven. Spent a fair bit of time trying to figure out what was going on before I checked the number of branches on the master repository.

**Expected behavior**
That a fork of the ES repository will continue to point to master, being the latest point in ES development, and that all commits will be applied to master.

Given that the release point is, by definition, in the past, it should be the branch off of master that remains static (or only gets fixes) and master is the working branch that continues forward.

Reasons:

- All github documentation talks about synchronizing to master. Git is difficult enough for non-coder content-creators to deal with as it is, let alone when things start diverging from the simplest possible behavior.
- Everyone engaged in content creation or coding is going to want to synchronise with the latest version of the repository, and by default that has been master. Having to switch to an alternate branch for everyone is going to be a time consuming task and probably a source of endless annoyance for the various support platforms.
- In terms of development, the release is basically frozen with little-to-no changes, whereas development continues. From my perspective, it makes more sense to branch off the release, because no-one except the Devs will ever need to refer to it again, rather than the development stream which *everyone* will need to deal with regularly.
- We have a largely non-coding community. It makes the most sense to optimize the repository to minimize the need for git hand-holding by the coders, and maximize the simplicity for the content creating non-coders. Basically, have the frozen branch for the release, and leave the "master" branch for development, since that means that *nothing changes* for all the people who would otherwise just get confused.
- Keep It Simple: No-one except the developpers and maintainers need to have anything to do with the release branch, whereas potentially hundreds of people need to point to the current development branch. Stick with the simplest solution that needs the fewest people to adapt to it to avoid confusing and/or losing content creators.

**Desired Result**
That the master branch continues to be the active development branch, and the release point should be a branch off of it.

**Comment by Terin** [comment]
Definitely agree that master should be 0.9.15+, and we should put 0.9.14 on a separate branch. The rationale behind what we have now, from my understanding, is that it'll minimize complications for when MZ decides to release 0.9.14 proper. However, there are a handful of reasons why this isn't ideal:

A) Build actions aren't currently running on the branch. We're not able to get the error-checking completed, nor will the continuous build update on the release page, or for something like ESLauncher 2.
B) All the confusions listed here.
C) MZ seems to not give Endless Sky any attention until he has something like an entire weekend day set aside. For 0.9.13, he ran into a pretty obscure bug building the game, and was able to tackle that. Any confusion he'd have about picking branches seems like something he could overcome much quicker than what happened last time.

With all that, I definitely think it should be swapped before we get too far.

**Comment by petervdmeer** [comment]
Calling this a bug doesn't seem appropriate; Releases have always been made from the master branch. We can (and did) request for that to change, but the decision to actually change the release branch lies with the person creating the (Steam) release.

Can you change this from a bug-report to a process-feature-request? Filing a bug-report suggests that creation of the v0.9.15 branch was wrong, but I'm actually happy that the devs are providing a branch where we can already make progress for the next release while master is frozen. The alternative of just waiting for master to become unfrozen or the other alternative of disturbing the Steam release workflow (which is already heavily affected by the [bus-factor](https://github.com/endless-sky/endless-sky/issues/3723)) both seem less desirable.

I know that Git can be quite intimidating and unforgiving to non-version-control-experts. We might want to start officially documenting the ways of working with v0.9.xx branches (GitHub already provides a [description on retargeting](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) ) if we plan to use v0.9.xx more in the future.

**Comment by Zitchas** [comment]
> Calling this a bug doesn't seem appropriate; Releases have always been made from the master branch. We can (and did) request for that to change, but the decision to actually change the release branch lies with the person creating the (Steam) release.

Reworded it to "Configuration Confusion."

> Can you change this from a bug-report to a process-feature-request? Filing a bug-report suggests that creation of the v0.9.15 branch was wrong, but I'm actually happy that the devs are providing a branch where we can already make progress for the next release while master is frozen. The alternative of just waiting for master to become unfrozen or the other alternative of disturbing the Steam release workflow (which is already heavily affected by the [bus-factor](https://github.com/endless-sky/endless-sky/issues/3723)) both seem less desirable.

> 
> I know that Git can be quite intimidating and unforgiving to non-version-control-experts. We might want to start officially documenting the ways of working with v0.9.xx branches (GitHub already provides a [description on retargeting](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) ) if we plan to use v0.9.xx branches more in the future.

I think that might be an understatement. I've been learning about it for a couple years now here, and I *still* can't even so much as squash a stack of my own commits. Trying to squash commits, change where a branch is pointing, revert commits, and a few other things of that sort, so far, universally resulted in a wrecked branch and me nuking the whole thing and re-establishing a new branch pointed wherever it needed to go with the changes copied in from an external-to-git backup copy. 

Likewise, I've seen specifically cite dealing with git *at all* as being a significant reason for not getting involved in contributing to vanilla. In fact, I've done a few PRs to implement fixes for various things over the years for people who did all the work of making the fix, but who couldn't make heads nor tails of git the way we have had it so far with just a single target to deal with. 

In other words, I consider it a bug-level item that the main repository needs to be kept as absolutely bare-bones simple as possible for the rank-and-file contributor. I seriously doubt that switching to a 0.9.14 branch will take anything more than a few seconds of thought for any of those with commit access to the repository. And actually, should *save* time for three of them, since the release branch would be simply broken off and frozen there, to be abandoned for MZ to deal with at his pleasure, while the master and the rest of the repository continues development.

**Comment by Pointedstick** [comment]
I think there's nothing left to do here, right?

I agree that git can be nasty and has a learning cliff. That said, this is *git*hub; ya gotta learn it sometime. :) I have been extensively involved in git-related onboarding  and it really is mostly about documentation, documentation, documentation. the workflow is baked into the system itself and the tools you use to interact with it.

In the end, git's gonna git.
